File: BaseDatosJava/ProductoDAOImpl.java


import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Statement;
import java.util.ArrayList;
import java.util.List;

public class ProductoDAOImpl implements IProductoDAO{

@Override
public boolean registrar(Producto producto){

boolean registrar = false;
Statement stm= null;
Connection con= null;

String sql="INSERT INTO producto VALUES(NULL,'"+producto.getcodigo_Producto()+"','"+producto.getnombre_Producto()+"','"+producto.getcategoria()+"','"+producto.getventa_Dia1()+"','"+producto.getventa_Dia2()+"','"+producto.getventa_Dia3()+"')";

try {

    con = Conexion.conectar();
    stm = con.createStatement();
    registrar = true;
    stm.close();
    con.close();

} catch (SQLException e){

    System.out.println("Error: Clase ProductoDAOImpl, metodo registrar ");
    e.printStackTrace();
}

return registrar;

}

@Override
public List<Producto>obtener(){
Connection co=null;
Statement stm= null;
ResultSet rs = null;

String sql="SELECT * FROM producto ORDER BY ID";
List<Producto> listaProducto =  new ArrayList<Producto>();

try{
    co = Conexion.conectar();
    stm = co.createStatement();
    rs= stm.executeQuery(sql);
    while(rs.next()){

     Producto c = new Producto();
     c.setid(rs.getInt(1));
     c.setcodigo_Producto(rs.getString(2));
     c.setnombre_Producto(rs.getString(3));
     c.setcategoria(rs.getString(4));
     c.setventa_Dia1(rs.getInt(5));
     c.setventa_Dia2(rs.getInt(6));
     c.setventa_Dia3(rs.getInt(7));
     
     listaProducto.add(c);

    }

    stm.close();
    rs.close();
    co.close();
}  catch(SQLException e) {

    System.out.println("Error: Clase ProductoDAOImpl, metodo obtener ");
    e.printStackTrace();
}
return listaProducto;
}

@Override
public boolean actualizar(Producto producto){
Connection connect=null;
Statement stm= null;
boolean actualizar = false;

String sql="UPDATE producto SET codigo_producto='"+producto.getcodigo_Producto()+"',nombre_producto='"+producto.getnombre_Producto()+"',categoria='"+producto.getcategoria()+"',venta_dia1='"+producto.getventa_Dia1()+"',venta_dia2='"+producto.getventa_Dia2()+"',venta_dia3='"+producto.getventa_Dia3()+"'"+"WHERE id="+producto.getid();


try{
    connect = Conexion.conectar();
    stm = connect.createStatement();
     stm.execute(sql);
     actualizar= true;
  }  catch(SQLException e) {

    System.out.println("Error: Clase ProductoDAOImpl, metodo actualizar ");
    e.printStackTrace();
}

return actualizar;
}


@Override
public boolean eliminar(Producto producto){
Connection connect=null;
Statement stm= null;
boolean eliminar = false;

String sql="DELETE FROM producto WHERE id="+producto.getid();

try{
    connect = Conexion.conectar();
    stm = connect.createStatement();
     stm.execute(sql);
     eliminar= true;
  }  catch(SQLException e) {

    System.out.println("Error: Clase ProductoDAOImpl, metodo eliminar ");
    e.printStackTrace();
}

return eliminar;
}


}

File: PDO POO PHP/ProductoDAOImpl.php
<?php

require_once("config.php");

class ProductoDAOImpl extends Conexiondb{

private $conexion;

private $id,$codigo_producto,$nombre_producto,$categoria,$venta_dia1,$venta_dia2,$venta_dia3;

public function __construct()
{
    $this->conexion=Conexiondb::conectar();

   

}

public function getListarProducto(){

    $declaracion= $this->conexion->prepare("SELECT * from producto order by id");

    $declaracion->execute();

    $declaracion->bindColumn("codigo_producto",$codigo);
    $declaracion->bindColumn("nombre_producto",$nombre);
    $declaracion->bindColumn("categoria",$categoria);
    $declaracion->bindColumn("venta_dia1",$venta_dia1);
    $declaracion->bindColumn("venta_dia2",$venta_dia2);
    $declaracion->bindColumn("venta_dia3",$venta_dia3);

    return $declaracion->fetchAll();

  

}

public function setInsertarProducto($codigo_producto,$nombre_producto,$categoria,
$venta_dia1,$venta_dia2,$venta_dia3){

    $declaracion = $this->conexion->prepare("INSERT into producto (codigo_producto,nombre_producto,
    categoria,venta_dia1,venta_dia2,venta_dia3) values(:codigo_producto,:nombre_producto,:categoria,
    :venta_dia1,:venta_dia2,:venta_dia3)");

    $declaracion->bindParam(":codigo_producto",$codigo_producto);
    $declaracion->bindParam(":nombre_producto",$nombre_producto);
    $declaracion->bindParam(":categoria",$categoria);
    $declaracion->bindParam(":venta_dia1",$venta_dia1);
    $declaracion->bindParam(":venta_dia2",$venta_dia2);
    $declaracion->bindParam(":venta_dia3",$venta_dia3);

    $declaracion->execute();
}


public function setActualizar($id,$codigo_producto,$nombre_producto,$categoria,
$venta_dia1,$venta_dia2,$venta_dia3){

$declaracion = $this->conexion->prepare("UPDATE producto set codigo_producto=:codigo_producto,
nombre_producto=:nombre_producto,
categoria=:categoria,venta_dia1=:venta_dia1,venta_dia2=:venta_dia2,venta_dia3=:venta_dia3
where id=:id");

$declaracion->bindParam(':id',$id);
$declaracion->bindParam(':codigo_producto',$codigo_producto);
$declaracion->bindParam(':nombre_producto',$nombre_producto);
$declaracion->bindParam(':categoria',$categoria);
$declaracion->bindParam(':venta_dia1',$venta_dia1);
$declaracion->bindParam(':venta_dia2',$venta_dia2);
$declaracion->bindParam(':venta_dia3',$venta_dia3);

$declaracion->execute();

}

public function getMostrarProducto($id){

    $declaracion = $this->conexion->prepare("SELECT * from producto where id='$id'");

$declaracion->execute();

$declaracion->bindColumn("id",$id);
$declaracion->bindColumn("codigo_producto",$codigo);
    $declaracion->bindColumn("nombre_producto",$nombre);
    $declaracion->bindColumn("categoria",$categoria);
    $declaracion->bindColumn("venta_dia1",$venta_dia1);
    $declaracion->bindColumn("venta_dia2",$venta_dia2);
    $declaracion->bindColumn("venta_dia3",$venta_dia3);

   return $declaracion->fetchAll();

}

public function setEliminar($id){

    $declaracion = $this->conexion->prepare("DELETE from producto where id=:id");


$declaracion->execute(array(":id" => $id));
}

}

?>






